Tests/SystemTests/NetworkConnectivityTests/Subscriber/Helper/SubscriberMonitor.swift

```diff
@@ -0,0 +1,382 @@
+                        .store(in: &subscriptions)
+                }
+            }
+        }
+
+        // MARK: - Assertions
+
+        private enum AssertionError: Error {
+            case generic(message: String)
+        }
+
+        private func createGenericAssertionError(message: String) -> AssertionError {
+            let taggedMessage = logHandler.tagMessage(message)
+            return .generic(message: taggedMessage)
+        }
+
+        private func assertStateTransition(waiterContext: Blocking.WaiterContext) throws {
+            logHandler.logMessage(level: .debug, message: "Awaiting state transition to \(expectedState)", error: nil)
+
+            let publisher = sharedState.combineSubscriberDelegate.trackableStates
+                .compactMap { [failureStates, expectedState, logHandler] state in
+                    SubscriberMonitor.receive(
+                        state,
+                        failureStates: failureStates,
+                        expectedState: expectedState,
+                        logHandler: logHandler
+                    )
+                }
+
+            let success = try first(publisher, label: "Wait for state transition to \(expectedState)", waiterContext: waiterContext)
+            if (!success) {
+                throw createGenericAssertionError(message: "Wait for state transition to \(expectedState) did not result in success.")
+            }
+        }
+
+        /**
+         * Maps received `ConnectionState` to a success/fail/ignore outcome for this test.
+         */
+        private static func receive(_ state: ConnectionState, failureStates: Set<ConnectionState>, expectedState: ConnectionState, logHandler: InternalLogHandler) -> Bool? {
+            if failureStates.contains(state) {
+                logHandler.logMessage(level: .error, message: "(FAIL) Got state \(state)", error: nil)
+                return false
+            }
+
+            if state == expectedState {
+                logHandler.logMessage(level: .debug, message: "(SUCCESS) Got state \(state)", error: nil)
+                return true
+            }
+
+            logHandler.logMessage(level: .debug, message: "(IGNORED) Got state \(state)", error: nil)
+            return nil
+        }
+
+        private func assertPublisherResolution(waiterContext: Blocking.WaiterContext) throws {
+            guard let expectedPublisherResolution else {
+                logHandler.debug(message: "(SKIP) expectedPublisherResolution = nil", error: nil)
+                return
+            }
+
+            logHandler.debug(message: "(WAITING) expectedPublisherResolution = \(expectedPublisherResolution)", error: nil)
+
+            let lastPublisherResolution = try listenForExpectedPublisherResolution(waiterContext: waiterContext)
+
+            // TODO this seems a bit redundant – listenForExpectedPublisherResolution will either throw or return expectedPublisherResolution, won't it?
```

Comment:
@AndyTWF I was wondering if you had any thoughts on this — did I miss something?

Reply:
Yeah - it looks like first() will throw if it times out waiting for a value, so I think you can remove this check.

Reply:
Cool, will remove 👍